server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/examples/activeTerminals/RestServiceActiveTerminals.java

```diff
@@ -0,0 +1,75 @@
+package com.generalbytes.batm.server.extensions.extra.examples.activeTerminals;
+
+import com.generalbytes.batm.server.extensions.IApiAccess;
+import com.generalbytes.batm.server.extensions.ITerminal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Path("/")
+public class RestServiceActiveTerminals {
+    private static final Logger log = LoggerFactory.getLogger("batm.master.extensions.activeTerminals.ActiveTerminalsExtension");
+    private static long pingDelay = 300000; // 300000 ms = 5min
+
+    @GET
+    @Path("/terminals")
+    @Produces(MediaType.APPLICATION_JSON)
+    /**
+     * https://192.168.51.152:7743/extensions/example/active/terminals
+     * Returns list of active terminals ( active = pingDelay < 5 min ).
+     */
+    public Object terminals(@QueryParam("api_key") String apiKey) {
+        try {
+            IApiAccess api_key = ActiveTerminalsExtension.getExtensionContext().getAPIAccessByKey(apiKey);
+            if (api_key != null) {
+                List<ITerminal> terminals = getTerminalsByApiKey(api_key);
+                return terminals;
+            }
+        } catch (Throwable e) {
+            log.debug("Wrong access to active terminals RestApi ", e);
+        }
+        return "wrong api key";
```

Comment:
Well instead of returning this string it would be better to throw some exception to produce 401 Unauthorized response